src/arith/num_big.hpp

```
@@ -3,63 +3,221 @@
+        int count = 0;
+        for (uint64_t mask = uint64_t(1) << 63; mask != 0 && !(lo & mask); mask >>= 1) {
+            count++;
+        }
+        return 64 + count;
+#endif
+    }
+    return 128;
+}
+
+// Parse a decimal string into Int128.
+inline Int128 parse_int128(const std::string& s) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)